### DISCUSSION TITLE
How to set multiple administration in Mobi

### DISCUSSION THREAD

**Discussion by AoleZheng** [Open-Systems-Pharmacology/Forum#1861] [Questions & Problems]
Hi!  

In my project, I have built a pregnancy population model in Mobi with vaginal administration. I was able to successfully fit the model to single-dose observed data by adding some parameters in the "Event" module. Now, I would like to validate the model with multiple-dose observed data. In the clinical study, the dosing regimen is 100 mg BID for 10 days.

However, creating 20 separate "application" events and setting parameters for each one is tedious and error-prone. It is also difficult for the following fitting process.

So I wonder if there was a more simple way to set mutiple administration in Mobi. Such as setting interval Time and Number of dose to integrate all administration in one "application".

**Comment by Sulafa93** [comment]
Hi! 

I have the same question. Did you find a solution? I’m guessing you set up the administration protocol in PK-Sim and then sent it to MoBi.

Thanks!

**Comment by SCZ1174** [comment]
You can create an arbitrary simulation in PKsim using the “100 mg BID for 10 days” Administration Protocols and save the pkml file. Then load the pkml in Mobi, remove all other modules associated with it and use it when creating the simulation.

**Reply by rwmcintosh** [reply]
That's a good tip!

You can also load the Events Building Block directly from a simulation PKML by right-clicking on a module and selecting "Load Building Blocks..."
![image](https://github.com/user-attachments/assets/7cca1bc9-833f-41ed-bf4f-ff5a400c6a24)

Then point to the simulation PKML that has the events you need, and it will import only the events from the PKML.

That's a general feature of loading PKML so it should also work in versions before V12.

**Reply by Sulafa93** [reply]
Thank you! But If I'm editing the event in MoBi (e.g., using a formula) and I have 10 applications, is it possible to modify just application 1 and then apply the changes to the remaining 9 applications without manually editing each one individually?

**Reply by Harris728** [reply]
I have the same question, hahaha

**Reply by AoleZheng** [reply]
> Thank you! But If I'm editing the event in MoBi (e.g., using a formula) and I have 10 applications, is it possible to modify just application 1 and then apply the changes to the remaining 9 applications without manually editing each one individually?

I have a compromise solution for this.  For modifying some formulation parameters such as dissolution shape and dissolution time, you can build an "Event Group" which can gather all applications. All formulation parameters you want to modify are added in "Parameters" of "Event Group". Then in subordinate applications, set the correct pathway and formulations. As such, once you change the parameters in "Event Group", all applications will be changed.

**Reply by SCZ1174** [reply]
This is a good method.
As a side note, the parameter you add to the index can be anywhere, as long as your “Path” is set correctly.
As I understand, this means that you also need to manually change the parameter “Formula type” from Constant to Formula in all applications, right? But there really isn't an ideal “one-click-application” feature at the moment, haha.

**Reply by AoleZheng** [reply]
> This is a good method. As a side note, the parameter you add to the index can be anywhere, as long as your “Path” is set correctly. As I understand, this means that you also need to manually change the parameter “Formula type” from Constant to Formula in all applications, right? But there really isn't an ideal “one-click-application” feature at the moment, haha.

As far as I know,  the "Path" can't cite the same directory level, so just check that the parameters were added in a higher-level directory. In my case, I tend to set an application for a template and load this application. I just need to change the "Start Time" for dosing.  It is easier for fitting parameters, although it is still tedious to set multiple administrations.
